0_json_functions.R

```diff
@@ -348,7 +348,7 @@ export_audit_invalid_data <- function(portfolio_total_, export_path_full) {
   invalidsecurties <- toJSON(portfolio_total_, dataframe = c("columns"))
 
   write(invalidsecurties, file = paste0(export_path_full, ".json"))
-  readr::write_csv(portfolio_total_, path =  paste0(export_path_full, ".csv"))
+  readr::write_csv(portfolio_total_, paste0(export_path_full, ".csv"))
```

Comment:
Despite the change in name, the argument `path/file` has always remained in the second position. This solution is backward compatible, consistent with other uses of `write_csv()` in this repo, and light-weight.

Reply:
hmmm.... I don't love it, but it is an ingenious solution to working in both circumstances 👍🏻

Reply:
Yeah, I know. If you prefer I can implement `readr_new_interface()` as shown above. Let me know what you'd be happy to approve so we get this out of the way.

Reply:
Ugh, I don't know.... I guess checking the version is the most proper solution and avoids someone "fixing" this later and breaking it again without leaving annoying comments in the code everywhere this happens @AlexAxthelm what's the least amount of "tech debt" we can create from this situation?

Reply:
Hmm. I'm a big fan of named arguments, but this does have some elegance to it. 🤔 

If we were only committed to supporting the "web version", I'd say make the change and live with it. But if our goal is to support a wide range of platforms, I think putting the conditional code in is best. And that opens the door for us to start including new features that are dependent upon certain package versions. With that we could start offering messages like:
> Installed version of `readr` is 1.4.0
> Exporting pictures of turtles is not supported with this version of `readr`. To enable this, please upgrade `readr` to at least version 1.6.3
> `<install.packages(blahblahblah)>`

Reply:
So I guess my opinion is: put the conditionals in, but don't feel compelled to put in too much work to support old versions

Reply:
I'd prefer to support more than just the web version because in my long term ideal, there would only be one version, and that will need to work in multiple contexts. Maybe if/when this gets package-ized one day, we can enforce stricter dependency rules/versions globally.

So @maurolepore, sorry for all the run around, but let's go for what you suggested with `readr_new_interface()`.

Reply:
rather than having a `readr_new_interface()` function, could we have the conditional check the version number in place? Checking the version isn't that expensive (computationally), and it avoids having to define yet another function down the line, if the API changes again (`readr_even_newer_interface()`??)

```r
# file argument to write_csv was introduced in readr x.y.z
if (packageVersion("readr") >= "x.y.z") {
  readr::write_csv(file = "path/to/file")
} else {
  readr::write_csv(path = "path/to/file")
}
```

Reply:
And of course, we'll need to make sure that we have a way to test the old versions too.

Reply:
@AlexAxthelm that's what I would probably naturally lean toward, but I'm almost certain @maurolepore will prefer making a mini function for it since it will need to be repeated more than once, and I can at least understand the value in that. Maybe we should be making a `save_write_csv()` function that handles all of it? like...
``` r
save_write_csv <- function(data, file, ...) {
  if (packageVersion("readr") >= "1.4.0") {
    readr::write_csv(data, file = file, ...)
  } else {
    readr::write_csv(data, path = file, ...)
  }
}

save_write_csv(data, file = "path/to/file.csv")
```

Reply:
Oh, I thought this all already inside a function. That seems like a good path to me. We should also start testing against other package versions, to figure out what the minimum version of each of our dependencies already is. But that would require _some_ level of automated testing.